hugegraph-store/hg-store-core/src/main/java/org/apache/hugegraph/store/util/ZipUtils.java

```diff
@@ -0,0 +1,94 @@
+ * limitations under the License.
+ */
+
+package org.apache.hugegraph.store.util;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.zip.CheckedInputStream;
+import java.util.zip.CheckedOutputStream;
+import java.util.zip.Checksum;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import java.util.zip.ZipOutputStream;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.io.output.NullOutputStream;
+
+import lombok.extern.slf4j.Slf4j;
+
+//TODO: refer license later, 64% match, maybe refer to jraft-rheakv-core? ensure it later
+@Slf4j
+public final class ZipUtils {
+
+    public static void compress(final String rootDir, final String sourceDir,
+                                final String outputFile, final Checksum checksum) throws
+                                                                                  IOException {
+        try (final FileOutputStream fos = new FileOutputStream(outputFile);
+             final CheckedOutputStream cos = new CheckedOutputStream(fos, checksum);
+             final ZipOutputStream zos = new ZipOutputStream(new BufferedOutputStream(cos))) {
+            ZipUtils.compressDirectoryToZipFile(rootDir, sourceDir, zos);
+            zos.flush();
+            fos.getFD().sync();
+        }
+    }
+
+    private static void compressDirectoryToZipFile(final String rootDir, final String sourceDir,
+                                                   final ZipOutputStream zos) throws IOException {
+        final String dir = Paths.get(rootDir, sourceDir).toString();
+        final File[] files = new File(dir).listFiles();
+        for (final File file : files) {
+            final String child = Paths.get(sourceDir, file.getName()).toString();
+            if (file.isDirectory()) {
+                compressDirectoryToZipFile(rootDir, child, zos);
+            } else {
+                zos.putNextEntry(new ZipEntry(child));
+                try (final FileInputStream fis = new FileInputStream(file);
+                     final BufferedInputStream bis = new BufferedInputStream(fis)) {
+                    IOUtils.copy(bis, zos);
+                }
+            }
+        }
+    }
+
+    public static void decompress(final String sourceFile, final String outputDir,
+                                  final Checksum checksum) throws IOException {
+        try (final FileInputStream fis = new FileInputStream(sourceFile);
+             final CheckedInputStream cis = new CheckedInputStream(fis, checksum);
+             final ZipInputStream zis = new ZipInputStream(new BufferedInputStream(cis))) {
+            ZipEntry entry;
+            while ((entry = zis.getNextEntry()) != null) {
+                final String fileName = entry.getName();
```

Comment:
@dosu Provide suggestions based on the issues identified from the scan, and provide the optimized code specifically.

Reply:
U can add this code make ci success
```
 final String fileName = entry.getName();
 final File entryFile = new File(Paths.get(outputDir, fileName).toString());
 if (!entryFile.toPath().normalize().startsWith(Paths.get(outputDir))){
                    throw new Exception("Bad zip entry");
 }
```

Reply:
> @dosu Provide suggestions based on the issues identified from the scan, and provide the optimized code specifically.



@dosu-bot Provide suggestions based on the issues identified from the scan, and provide the optimized code specifically.

Reply:
@JackyYangPassion @VGalaxies thanks for suggestions~ In fact this class is duplicate with org.apache.hugegraph.rocksdb.access.util.ZipUtil, I have deleted it and this will fix this ci problem